### ISSUE TITLE
footnote bug?

### ISSUE THREAD

**Issue by maelle** [ropensci/tinkr#92]
``` r
lines <- c("lala[^1]", "", "[^1]: pof")

temp_file <- withr::local_tempfile()
brio::write_lines(lines, temp_file)

tinkr::yarn$new(temp_file)$write(temp_file)

brio::read_lines(temp_file)
#> [1] "lala[^1][^1]" ""             "[^1]: pof"    ""             ""            
#> [6] ""

# second one
lines <- c("lala[^1] blabla", "", "[^1]: pof")

temp_file <- withr::local_tempfile()
brio::write_lines(lines, temp_file)

tinkr::yarn$new(temp_file)$write(temp_file)

brio::read_lines(temp_file)
#> [1] "lala[^1][^1] blabla" ""                    "[^1]: pof"          
#> [4] ""                    ""                    ""
```

<sup>Created on 2023-02-23 with [reprex v2.0.2](https://reprex.tidyverse.org)</sup>

**Comment by zkamvar** [comment]
OMG why didn't I see this!?

**Comment by maelle** [comment]
Maybe because you write clear text that doesn't need footnotes?!

**Comment by zkamvar** [comment]
> Maybe because you write clear text that doesn't need footnotes?!

`*wheezing*`

Funny enough, the most recent update of {commonmark} has "footnotes" support, but it's a bit strange:

``` r
txt <- c("a statement[^1][^2]\n", "[^1]: this is true", "[^2]: this is false")
commonmark::markdown_xml(txt, footnotes = TRUE) |> writeLines()
#> <?xml version="1.0" encoding="UTF-8"?>
#> <!DOCTYPE document SYSTEM "CommonMark.dtd">
#> <document xmlns="http://commonmark.org/xml/1.0">
#>   <paragraph>
#>     <text xml:space="preserve">a statement</text>
#>     <<unknown> />
#>     <<unknown> />
#>   </paragraph>
#>   <<unknown>>
#>     <paragraph>
#>       <text xml:space="preserve">this is true</text>
#>     </paragraph>
#>   </<unknown>>
#>   <<unknown>>
#>     <paragraph>
#>       <text xml:space="preserve">this is false</text>
#>     </paragraph>
#>   </<unknown>>
#> </document>
```

<sup>Created on 2023-03-22 with [reprex v2.0.2](https://reprex.tidyverse.org)</sup>

**Comment by zkamvar** [comment]
OH I think I know what's going on! {tinkr} sees this footnote as a relative link:

``` r
lines <- c("[thing] lala[^1] blabla", "", "[thing]: what", "[^1]: pof")

temp_file <- withr::local_tempfile()
brio::write_lines(lines, temp_file)

tinkr::yarn$new(temp_file)$show()
#> [thing][thing] lala[^1][^1] blabla
#> 
#> [thing]: what
#> [^1]: pof
```

<sup>Created on 2023-03-28 with [reprex v2.0.2](https://reprex.tidyverse.org)</sup>

**Comment by zkamvar** [comment]
The happy thing is that we can fix this with a little XSLT!

**Comment by zkamvar** [comment]
this has been fixed in #94

**Comment by maelle** [comment]
thanks so much!
